modules/view_establishment/predicates.py

```diff
@@ -0,0 +1,344 @@
+        Returns a set of processors that has the view pair as node j
+        (and phase if passed)
+        and if node j is not stale.
+        """
+        raise NotImplementedError
+
+    def transit_adopble(self, node_j, phase, mode):
+        """Methoid description.
+
+        Checks if 3f+1 processors have reported to remain in
+        or transiting to the phase.
+        """
+        raise NotImplementedError
+
+    def transit_set(self, node_j, phase, mode):
+        """Method description.
+
+        If called with mode = "REMAIN"
+        - Returns the set of nodes that will support
+        to remain in the same view as node j.
+        If called with mode = "FOLLOW
+        - Returns the set of nodes that will support
+        to follow to a new view or to a view change.
+        """
+        raise NotImplementedError
+
+    def transition_cases(self, node_j, vpair, phase, mode):
+        """Method description.
+
+        Examines three cases.
+        mode = "REMAIN"
+        - Returns true if the current view of node j is the same as vpair.next.
+        mode = "FOLLOW, phase = 0
+        - Returns true if vpair.next is the consecutive view of the current
+        view of node j.
+        mode = "FOLLOW, phase = 1
+        - Returns true if vpair.current is the same as the next view of node j
+        """
+        raise NotImplementedError
+
+    def adopt(self, vpair):
+        """Assign the current view pair as vpair."""
+        raise NotImplementedError
+
+    def establishable(self, phase, mode):
+        """Method description.
+
+        Checks if 4f+1 nodes are willing to move to a view change (phase 0)
+        or to a new view (phase 1).
+        """
+        raise NotImplementedError
+
+    def establish(self):
+        """Update the current view in the view pair to the next view."""
+        raise NotImplementedError
+
+    def next_view(self):
+        """Updates the next view in the view pair to upcoming view."""
+        raise NotImplementedError
+
+    def reset_v_change(self):
+        """The node is no longer in a view change."""
+        raise NotImplementedError
+
+    # Interface functions
+    def need_reset(self):
+        """True if the replication module requires a reset."""
+        return (self.stale_v() and self.resolver.execute(
+                module=Module.REPLICATION_MODULE,
+                func=Function.REPLICA_FLUSH
+                ))
+
+    def reset_all(self):
+        """Reset all modules."""
+        self.views = [self.RST_PAIR for i in range(self.number_of_nodes)]
+        self.view_module.init_module()
+        self.resolver.execute(
+            module=Module.REPLICATION_MODULE,
+            func=Function.REP_REQUEST_RESET
+        )
+        return("Reset")
+
+    def view_change(self):
+        """A view change is required."""
+        self.vChange = True
+
+    def get_view(self, node_j):
+        """Returns the most recent reported view of node_j."""
+        if (node_j == self.id or
```

Comment:
Should be `if (node_jd == self.id) and ...`

Reply:
Done, updated test accordingly